examples/WriteText/WriteText.ino

```diff
@@ -0,0 +1,123 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************
+ */
+
+ /**
+ ******************************************************************************
+* How to use this sketch
+*
+* This sketch uses the I2C interface for communicate with the NFC device.
+* It writes an NFC tag type Text and reads this same tag.
+* Check the pinout and the I2C interface used for communication.
+* Choose the content of the text by changing the content of text_write.
+*
+* We can see on the monitor window the message "Sytstem init done!" when the
+* NFC module is started and ready. Next, the tag is written, we wait few seconds,
+* we read the same tag and print it on the monitor window.
+*
+* You can test this application by connecting it with your smartphone.
+* If you want to read tag in RF mode (with smartphone) unplug the USB on the board.
+* The board should be power-off, or RF mode will not be available.
+*
+* You can write a Text tag with your smartphone.
+* On Android, donwload the NFC Tools.
+* Then start the app, check if NFC is activated on your smartphone.
+* Choose "ecrire" and then select the Text tag.
+* You can also read the text tag with the same app.
+ ******************************************************************************
+ */
+
+#include <Wire.h>
+#include <ArduinoSTL.h>
+#include <M24SR.h>
+
+#define SerialPort  Serial
+#define I2C2_SCL    PB10
+#define I2C2_SDA    PB11
+
+const uint8_t M24SR_ADDR=0xAC;
+const int     DEFAULT_GPO_PIN=PE4;
+const int     DEFAULT_RF_DISABLE_PIN=PE2;
+
+TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);
+
+uint8_t done = 0;
+M24SR nfcTag(M24SR_ADDR, &dev_i2c, NULL, DEFAULT_GPO_PIN, DEFAULT_RF_DISABLE_PIN);
+
+/**
+ * Write a Ndef Txt message, wait and read the message from
+ * the Nfc tag.
+ *
+ */
+void setup() {
+  // Initialize serial for output.
+  SerialPort.begin(9600);
+
+  // Initialize I2C bus.
+  dev_i2c.begin();
+
+  nfcTag.begin(NULL);
+
+  SerialPort.println("System Init done: !");
+}
+
+void loop() {
+  const char *text_write = "hello world welcome on earth";     // Text to write in the tag
+  char text_read[255];                                         // Text read in the tag
+
+  if(!done) {
+    delay(1000);
+    //write the txt message
+    if(nfcTag.writeTxt(text_write))
+    {
+      SerialPort.println("success write");
+    }
+    else
+    {
+      SerialPort.println("fail write");
+    }
+    delay(1000);
+    //read the txt message and print it
+    nfcTag.readTxt(text_read);
+    SerialPort.println("Message content :");
+    SerialPort.println(text_read);
```

Comment:
Need a strcpy to compare to the written message.

In my case, the read message is always empty... Perhaps I missed something in the way to test the NFC, but even in this case, we should get an error.

Reply:
One other change, is to perform the test in the setup as it is done only one

Reply:
@VVESTM 

There is a problem with the library. My release was ready since the last week before the PR of the I2C on IT. So it broke the library communication.
I must find a patch...

Reply:
Sounds strange that the I2C IT broke it.

Reply:
Perhaps it is a mismatch between i2c working with IT now and i2c buffer size. There are still I2C_TXRX_BUFFER_SIZE defined to 32 for example...
(I have tried to set it to 255 but it is not working).

Then, as mentioned by @fpistm, it can be good to have something more flexible. (Avoid to use a big buffer if not needed)

Reply:
I confirm the problem is in the I2C layer. It is not the buffer size the problem.

From the datasheet of the module we can read:
>A restart during a command and response exchange is not supported by the M24SR64-Y.
After a command, the I2C host can execute a Polling sequence to determine when the 
response is available.
Polling sequence: Loop on  START (S) + DeviceSelect with RW=0 + read NACK/ACK + STOP(P)
The response is available as soon as the M24SR64-Y sends an ACK ( host read will a '0').

That means the I2C goes in READY state but with AF (Address failure) error when it is waiting a response from the module because the module returns NACK. As the error code is not checked then the Wire library returns OK to the NFC driver but the module it is not ready yet!

The quick solution is to check the errorCode parameter in `i2c_master_read()`. But I saw other problems in the twi.c file. I will open an issue to expose to you those problems. (And I know I validated the PR but too quickly I think).

Reply:
My 2 cents: 
I think it is ok to exit i2c_master_read as soon as I2C READY internal state is reached
To check whether device is ready before sending a new command, the below API shall be used:
 i2c_status_e i2c_IsDeviceReady(i2c_t *obj, uint8_t devAddr, uint32_t trials)
It is up to the device driver to use this API when needed (as recommended by spec) before sending a new I2C command. I have seen EEPROM where is sometimes takes 5 or 6 checks before the device is actually ready ...